lib/rules/utils/ast-utils.js

```diff
@@ -2311,6 +2311,114 @@ module.exports = {
         return node.type === "TemplateLiteral" && node.expressions.length === 0;
     },
 
+    /**
+     * Determines whether the existing curly braces around the single statement are necessary to preserve the semantics of the code.
+     * The braces, which make the given block body, are necessary in either of the following situations:
+     *
+     * 1. The statement is a lexical declaration.
```

Comment:
This isn't checked by this function. We could either get the check back here (although it isn't necessary for the `no-lonely-if` rule) to make the function more generic, or remove this comment and rename this function to better describe that it only checks for one specific case (if-else pairing). I'd vote for the former. What do you think?

Reply:
Sounds good, done!